web/src/recent_view_data.ts

```diff
@@ -1,16 +1,41 @@
-import type {Message} from "./message_store";
+import assert from "minimalistic-assert";
+
+import type {DisplayRecipientUser, Message} from "./message_store";
 import * as people from "./people";
-import {get_key_from_message} from "./recent_view_util";
+import {get_key_from_conversation_data} from "./recent_view_util";
+import * as unread from "./unread";
 
 export type ConversationData = {
-    last_msg_id: number;
     participated: boolean;
-    type: "private" | "stream";
-};
+    unread_count: number;
+    latest_mesage_timestamp: number;
+    last_msg_id: number; // should not be used for fetching messages
```

Comment:
There are two places this ended up still being used for fetching messages. One is at the end of `get_focused_row_message`, but that's fine because it can return `undefined`. The other is generating`last_msg_url` in `format_conversation`, which I didn't create a fallback for because it would require a bunch more refactoring of `by_conversation_and_time_url` and the functions it calls.

Reply:
Cool, I think we should add a `TODO` there to reduce the change of us forgetting to take care of them.